src/rpc/client.cpp

```diff
@@ -325,69 +389,109 @@ static UniValue Parse(std::string_view raw)
+}
 
-    /** Return arg_value as UniValue, and first parse it if it is a non-string parameter */
-    UniValue ArgToUniValue(std::string_view arg_value, const std::string& method, int param_idx)
-    {
-        return members.count({method, param_idx}) > 0 ? Parse(arg_value) : arg_value;
-    }
+const CRPCConvertParam* FromName(std::string_view method, std::string_view name)
+{
+    auto it = std::ranges::find_if(vRPCConvertParams, [&](const auto& p) {
+        return p.methodName == method && p.paramName == name;
+    });
 
-    /** Return arg_value as UniValue, and first parse it if it is a non-string parameter */
-    UniValue ArgToUniValue(std::string_view arg_value, const std::string& method, const std::string& param_name)
-    {
-        return membersByName.count({method, param_name}) > 0 ? Parse(arg_value) : arg_value;
-    }
-};
+    return it == std::end(vRPCConvertParams) ? nullptr : &*it;
+}
+} // namespace rpc_convert
 
-CRPCConvertTable::CRPCConvertTable()
+static UniValue ParseParam(const CRPCConvertParam* param, std::string_view raw)
 {
-    for (const auto& cp : vRPCConvertParams) {
-        members.emplace(cp.methodName, cp.paramIdx);
-        membersByName.emplace(cp.methodName, cp.paramName);
-    }
+    // Only parse parameters which have the JSON format; otherwise, treat them as strings.
+    return (param && param->format == ParamFormat::JSON) ? Parse(raw) : UniValue(std::string(raw));
 }
 
-static CRPCConvertTable rpcCvtTable;
 
+/**
+ * Convert command lines arguments to params object when -named is disabled.
+ */
 UniValue RPCConvertValues(const std::string &strMethod, const std::vector<std::string> &strParams)
 {
     UniValue params(UniValue::VARR);
 
-    for (unsigned int idx = 0; idx < strParams.size(); idx++) {
-        std::string_view value{strParams[idx]};
-        params.push_back(rpcCvtTable.ArgToUniValue(value, strMethod, idx));
+    for (std::string_view s : strParams) {
+        params.push_back(ParseParam(rpc_convert::FromPosition(strMethod, static_cast<int>(params.size())), s));
     }
 
     return params;
 }
 
+/**
+ * Convert command line arguments to params object when -named is enabled.
+ *
+ * The -named syntax accepts named arguments in NAME=VALUE format, as well as
+ * positional arguments without names. The syntax is inherently ambiguous if
+ * names are omitted and values contain '=', so a heuristic is used to
+ * disambiguate:
+ *
+ * - Arguments that do not contain '=' are treated as positional parameters.
+ *
+ * - Arguments that do contain '=' are assumed to be named parameters in
+ *   NAME=VALUE format except for two special cases:
+ *
+ *   1. The case where NAME is not a known parameter name, and the next
+ *      positional parameter requires a JSON value, and the argument parses as
+ *      JSON. E.g. ["list", "with", "="].
+ *
+ *   2. The case where NAME is not a known parameter name and the next
+ *      positional parameter requires a string value. E.g. "==label==".
```

Comment:
In commit "rpc: Handle -named argument parsing where '=' character is used" (03068e67cd9a62aa9105ca6a4c971b6261a783c5)

Could change "==label==" to "my=wallet" to be consistent with example below. The "my=wallet" example is probably clearer anyway since there is only one `=` and it's in the middle of the string, and wallet names are probably more familiar than labels.